x-pack/legacy/plugins/snapshot_restore/public/app/components/policy_form/steps/step_settings.tsx

```diff
@@ -67,11 +67,11 @@ export const PolicyStepSettings: React.FunctionComponent<StepProps> = ({
   // State for using selectable indices list or custom patterns
   // Users with more than 100 indices will probably want to use an index pattern to select
   // them instead, so we'll default to showing them the index pattern input.
-  const [selectIndicesMode, setSelectIndicesMode] = useState<'list' | 'custom'>(
+  const [selectIndicesMode, setSelectIndicesMode] = useState<'custom' | 'list'>(
     typeof config.indices === 'string' ||
       (Array.isArray(config.indices) && config.indices.length > 100)
-      ? 'custom'
```

Comment:
I don't think this is quite what we want. If config.indices > 100, we should still default to the `custom` view.

Instead, we also need a way to detect if the user's selected policy indices - `config.indices` - contains any index patterns. If it does, then we should default to the `custom` view. One way this might be achieved is by comparing `indices` with `config.indices`.  For example (_not tested_):

```
const selectedIndices = indices.filter(index => config.indices.includes(index));
const showListView = selectedIndices.length === config.indices.length; // if the length is the same, we can assume the user only defined indices (not index patterns) and we show the `list` view.
```

Reply:
I added the above to the config.indices > 100 condition

```
// if the length is the same, we can assume the user only defined indices (not index patterns) and we show the `list` view.
  const selectedIndices = indices.filter(index => config.indices.includes(index))
  const showListView = selectedIndices.length === config.indices.length; 
  
// State for using selectable indices list or custom patterns
  // Users with more than 100 indices will probably want to use an index pattern to select
  // them instead, so we'll default to showing them the index pattern input.
  const [selectIndicesMode, setSelectIndicesMode] = useState<'list' | 'custom'>(
    typeof config.indices === 'string' ||
      (Array.isArray(config.indices) && config.indices.length > 100 && !showListView)
      ? 'custom'
      : 'list'
  );
```
Is that what you're referring to?

Reply:
> I don't think this is quite what we want. If config.indices > 100, we should still default to the >custom view.

>Instead, we also need a way to detect if the user's selected policy indices - config.indices - >contains any index patterns. If it does, then we should default to the custom view. One way >this might be achieved is by comparing indices with config.indices. For example (not tested):

@alisonelizabeth from my testing it looks like adding just `indices.filter(index => config.indices.includes(index))` already detect if the user's selected policy indices - config.indices - contains any index patterns and defaults to the custom view. 

For example:
```
  const [selectIndicesMode, setSelectIndicesMode] = useState<'list' | 'custom'>(
    typeof config.indices === 'string' ||
      (Array.isArray(config.indices) && config.indices.length > 100 && indices.filter(index => config.indices.includes(index));)
      ? 'custom'
      : 'list'
  );

```

Reply:
@alisonelizabeth @cjcenizal I've written a truth table to illustrate my understanding of the input and output


Row | Policy contain - index pattern | Default to 
------------ | ------------- | -------------
1 | Custom  | Custom pattern
2 | Built-in (system) | Select indices
3 | Custom & Built-in | Custom pattern
4 | Aliases | Custom pattern
5 | greater than 100 | Custom pattern

I think with this github ticket I"m trying to implement 1) and 4) . The other rows are implemented already. If not, I"ll work on implementing it too. Just want your input.

Reply:
@jkelastic this is great! Thanks for putting this together.

Can you clarify what you mean by "existing" and "non-existing"? I'm also not sure I understand the difference between "custom" and "non-existing". 

Another case that I don't think is included in the table - if the user doesn't specify any indices for a policy, I believe the default behavior is for a snapshot to be created with all open and started indices.

Reply:
@alisonelizabeth by "existing" I mean the indices in the policy are in `GET _cat/indices` and "non-existing` isn't . It looks like we can create a policy from DevTools using `PUT` with `non-existing` indices e.g. data-* and important  doesn't exist in `GET _cat/indices` , haven't been created yet

```
PUT /_slm/policy/daily-snapshots
{
  "schedule": "0 30 1 * * ?", 
  "name": "<daily-snap-{now/d}>", 
  "repository": "my_repository", 
  "config": { 
    "indices": ["data-*", "important"], 
    "ignore_unavailable": false,
    "include_global_state": false
  },
  "retention": {}
}
```

I'm not sure how to handle this. On one hand I don't think ES should allow us to createt policies with "non-existing" indices. On the othere hand, I see https://github.com/elastic/kibana/pull/41534 allows user to to create a policy with `non-existing` indices and then create the index

```
        description={
          <FormattedMessage
            id="xpack.snapshotRestore.restoreForm.stepLogistics.indicesDescription"
            defaultMessage="Creates new indices if they don’t exist. Restores existing indices
              if they are closed and have the same number of shards as the snapshot index."
          />
        }
```

Reply:
> Another case that I don't think is included in the table - if the user doesn't specify any indices for a policy, I believe the default behavior is for a snapshot to be created with all open and started indices.

@alisonelizabeth thanks, I've added the above to the table. I've put the default to custom pattern as `a snapshot to be created with all open and started indices.`  could have > 100 indices 

Row | Policy contain - index pattern | Default to 
------------ | ------------- | -------------
1 | Custom  | Custom pattern
2 | Built-in (system) | Select indices
3 | Custom & Built-in | Custom pattern
4 | Aliases | Custom pattern
5 | greater than 100 | Custom pattern
6 | non-existing  | Custom pattern - error index doesn't exist
7 | existing and non-existing  | Custom and/or Select Indices - X array of indices doesn't exist or   Custom and/or Select Indices - X array exist
8 | existing  | Custom and/or Select Indices list
9 | none  | Custom pattern

Reply:
@jkelastic thanks for clarifying this! I'm also assuming "custom" means a valid index pattern.

> I'm not sure how to handle this. On one hand I don't think ES should allow us to create policies with "non-existing" indices. 

I think this is the correct behavior. I took a look at the[ ES documentation](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/modules-snapshots.html) regarding indices and it says the following:

```
The list of indices that should be included into the snapshot can be specified using the indices parameter that supports multi index syntax. The snapshot request also supports the ignore_unavailable option. Setting it to true will cause indices that do not exist to be ignored during snapshot creation. By default, when ignore_unavailable option is not set and an index is missing the snapshot request will fail. 
```

--- 
Some feedback regarding the table:
- [correction] I think `7` would always be the custom pattern, since it contains non-existing indices
- [clarification] `8`  would default to the "custom pattern" if it contains an index pattern, otherwise it would be "Select indices"
- [correction] `9` should be the "Select indices" list with the toggle enabled
<img width="375" alt="Screen Shot 2020-01-31 at 8 45 01 AM" src="https://user-images.githubusercontent.com/5226211/73543911-fe49dc00-4405-11ea-8f28-32b8498322ae.png">

---
> I think with this github ticket I"m trying to implement 1) and 4) . The other rows are implemented already. If not, I"ll work on implementing it too. Just want your input.

I would recommend testing all the scenarios you've listed here, and make sure all of them behave as expected for this PR.

Reply:
@alisonelizabeth thanks for the feedback, I will work on this right away.

Row | Policy contain - index pattern | Default to 
------------ | ------------- | -------------
1 | Custom  | Custom pattern
2 | Built-in (system) | Select indices
3 | Custom & Built-in | Custom pattern
4 | Aliases | Custom pattern
5 | greater than 100 | Custom pattern
6 | non-existing  | Custom pattern - error index doesn't exist
7 | existing and non-existing  | Custom  pattern
8 | existing  | Custom and/or Select Indices list (default to the "custom pattern" if it contains an index pattern, otherwise it would be "Select indices")
9 | none  | "Select indices" list with the toggle enabled

Reply:
@alisonelizabeth Testing results, please let me know what you think

Row | Policy contain - index pattern | Should default to ? | Currently default to
------------ | ------------- | ------------- | -------------
1 | Custom  | Custom pattern | Select indices
2 | Built-in (system) | Select indices  | Select indices
3 | Custom & Built-in | Custom pattern | Select indices and doesn't check built-in index
4 | Aliases | Custom pattern
5 | greater than 100 | Custom pattern
6 | non-existing  | Custom pattern - error index doesn't exist | Select indices- ignore unavailable `true` doesn't create index
7 | existing and non-existing  | Custom  pattern | Select indices -same as 6)
8 | existing  | Custom and/or Select Indices list (default to the "custom pattern" if it contains an index pattern, otherwise it would be "Select indices") | Select indices - even with custom index pattern
9 | none  | "Select indices" list with the toggle enabled | Working as is

Reply:
@jkelastic I think this is getting close! Nice job working through this.

I see `aliases` and `greater than 100` do not have anything listed under the current behavior column. I’d imagine `aliases` is defaulting to “Select indices” (incorrect) and `greater than 100` is working as expected, but please verify.

> Select indices- ignore unavailable true doesn't create index

I’m not sure I follow this statement. I believe `ignore_unavailable` defaults to `false` in ES, so I don’t think we want to change any behavior related to this in the UI; it would be up to the user to set. (I'm not sure if that is what you intended by your comment, so please ignore if not 😄)

Everything else LGTM!

Reply:
> @jkelastic I think this is getting close! Nice job working through this.
> 
> I see `aliases` and `greater than 100` do not have anything listed under the current behavior column. I’d imagine `aliases` is defaulting to “Select indices” (incorrect) and `greater than 100` is working as expected, but please verify.
> 
> > Select indices- ignore unavailable true doesn't create index
> 
> I’m not sure I follow this statement. I believe `ignore_unavailable` defaults to `false` in ES, so I don’t think we want to change any behavior related to this in the UI; it would be up to the user to set. (I'm not sure if that is what you intended by your comment, so please ignore if not 😄)
> 
> Everything else LGTM!

@alisonelizabeth thanks for the input, Sorry, I still need to test `aliases` and `greater than 100` . For the `ignore_unavailable` defaults to `false` I meant to say even when I manually set it to `true` the index / indices aren't automatically created. It doesn't seem to work.

Reply:
@jkelastic I'm not sure it is supposed to create the index. You may need to confirm with an ES engineer, and adjust the help text in the UI if needed.

Per the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/modules-snapshots.html): 
> The snapshot request also supports the ignore_unavailable option. Setting it to true will cause indices that do not exist to be ignored during snapshot creation

Reply:
@alisonelizabeth thanks for the input, so when `ignore_unavailable` defaults to `false` it will causes indices that do not exist to not be ignored (opposite of `true`). Does that mean there would be an error? I don't see any error in DevTools or in the snapshot policy.